contracts/protocol/facets/SellerHandlerFacet.sol

```diff
@@ -209,110 +205,134 @@ contract SellerHandlerFacet is SellerBase {
 
         require(exists, NO_PENDING_UPDATE_FOR_ACCOUNT);
 
-        // Get storage location for seller
-        (, Seller storage seller, AuthToken storage authToken) = fetchSeller(_sellerId);
+        {
+            bool updateApplied;
 
-        for (uint256 i = 0; i < _fieldsToUpdate.length; i++) {
-            SellerUpdateFields role = _fieldsToUpdate[i];
+            // Get storage location for seller
+            (, Seller storage seller, AuthToken storage authToken) = fetchSeller(_sellerId);
 
-            // Approve admin update
-            if (role == SellerUpdateFields.Admin && sellerPendingUpdate.admin != address(0)) {
-                require(sellerPendingUpdate.admin == sender, UNAUTHORIZED_CALLER_UPDATE);
+            // Cache protocol lookups and sender for reference
+            ProtocolLib.ProtocolLookups storage lookups = protocolLookups();
```

Comment:
Since msgSender() is called a lot in this method, and each time it accesses storage (in the case of a meta tx), we should fetch it into a memory location here and reuse throughout. I realize you may be dealing with stack to deep here, so that may not be possible, but it would be good if we could do it.

Reply:
I had first created a local variable, but then I need to remove to fix stack to deep error

Reply:
There is a way to go around the stack too deep error - not by wrapping the bulk of the code in {}, but rather to do it already at the begining, since `exists` is used only there and it's later not needed.

If you start the `optInToSellerUpdate` with
```
        // Get seller pending update
        Seller storage sellerPendingUpdate;
        AuthToken storage authTokenPendingUpdate;
        {
            bool exists;
            (exists, sellerPendingUpdate, authTokenPendingUpdate) = fetchSellerPendingUpdate(_sellerId);

            // Be sure an update is pending
            require(exists, NO_PENDING_UPDATE_FOR_ACCOUNT);
        }
        
        bool updateApplied;
        address sender = msgSender();
```
And then replace all `msgSender()` with `sender` that will work.

As a side note, even if that was not possible, then better than getting rid of `address seller` would be to sacrifice `SellerUpdateFields role`. Still better to read multiple times from the memory than from the storage.

Reply:
that makes sense!